vpr/src/base/partition_region.cpp

```diff
@@ -26,20 +26,18 @@ bool PartitionRegion::is_loc_in_part_reg(t_pl_loc loc) {
     return is_in_pr;
 }
 
-PartitionRegion intersection(PartitionRegion& pr1, PartitionRegion& pr2) {
+PartitionRegion intersection(const PartitionRegion& cluster_pr, const PartitionRegion& new_pr) {
     /**for N regions in part_region and M in the calling object you can get anywhere from
      * 0 to M*N regions in the resulting vector. Only intersection regions with non-zero area rectangles and
      * equivalent subtiles are put in the resulting vector
      * Rectangles are not merged even if it would be possible
      */
     PartitionRegion pr;
```

Comment:
make another version of this function that changes cluster_pr instead of creating new pr and returning (that will be more efficient because it avoids extra copying)